pmd-core/src/main/java/net/sourceforge/pmd/PropertyDescriptor.java

```diff
@@ -32,12 +61,15 @@
      */
     String description();
 
+
     /**
-     * Denotes the value datatype.
+     * Denotes the value datatype. For multi value properties, this is not the
+     * List class but the list's component class.
      *
-     * @return Class
+     * @return Class literal of the value type
      */
-    Class<T> type();
+    Class<?> type();
```

Comment:
I see where you might have gotten confused. The javadoc for MultiValuePropertyDescriptor was a copy paste of that of SingleValuePropertyDescriptor, which I forgot to rewrite. It's fixed now, sorry about that